src/SwedbankPay.Sdk/SwedbankPayHttpClient.cs

```diff
@@ -0,0 +1,193 @@
+﻿using System;
+using System.Net;
+using System.Net.Http;
+using System.Text;
+using System.Threading.Tasks;
+
+using Microsoft.Extensions.Logging;
+
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using Newtonsoft.Json.Serialization;
+
+using SwedbankPay.Sdk.Exceptions;
+
+namespace SwedbankPay.Sdk
+{
+    internal class SwedbankPayHttpClient
+    {
+        private readonly HttpClient client;
+        private readonly ILogger logger;
+
+
+        internal SwedbankPayHttpClient(HttpClient client, ILogger logger)
+        {
+            this.client = client;
+            this.logger = logger;
+        }
+
+
+        internal async Task<string> GetRaw(string url)
+        {
+            var msg = new HttpRequestMessage(HttpMethod.Get, url);
+            msg.Headers.Add("Accept", "application/json");
+
+            HttpResponseMessage response;
+            try
+            {
+                response = await this.client.SendAsync(msg);
+            }
+            catch (HttpRequestException e)
+            {
+                throw new BadRequestException(e);
+            }
+            catch (TaskCanceledException te)
+            {
+                throw new ApiTimeOutException(te);
+            }
+
+            if (response.IsSuccessStatusCode)
+            {
+                var res = await response.Content.ReadAsStringAsync();
+                var res2 = JToken.Parse(res).ToString(Formatting.Indented);
+                this.logger.LogInformation(res2);
+                return res2;
+            }
+
+            return null;
+        }
+
+
+        internal async Task<TResponse> HttpGet<TResponse>(string url, Func<ProblemsContainer, Exception> onError)
+            where TResponse : new()
+        {
+            return await HttpRequest<TResponse>(HttpMethod.Get, url, onError);
+        }
+
+
+        internal async Task<TResponse> HttpPatch
+            <TPayLoad, TResponse>(string url, Func<ProblemsContainer, Exception> onError, TPayLoad payload)
+            where TResponse : new()
+        {
+            return await HttpRequest<TResponse>(new HttpMethod("PATCH"), url, onError, payload);
+        }
+
+
+        internal async Task<TResponse> HttpPost
+            <TPayLoad, TResponse>(string url, Func<ProblemsContainer, Exception> onError, TPayLoad payload)
+            where TResponse : new()
+        {
+            return await HttpRequest<TResponse>(HttpMethod.Post, url, onError, payload);
+        }
+
+
+        internal async Task<T> HttpRequest
+            <T>(string httpMethod, string url, Func<ProblemsContainer, Exception> onError, object payload = null)
```

Comment:
The only difference from the other HttpRequest() is that this takes a httpMethod input as a string. I can't see why this would be useful enough to warrant its own method when the caller could just wrap the string in a HttpMethod object themselves.